internal/plans/internal/planproto/planfile.proto

```diff
@@ -207,6 +212,36 @@ message OutputChange {
     bool sensitive = 3;
 }
 
+// ConditionType defines the type of condition block used to generate a result.
+enum ConditionType {
+	INVALID                = 0;
+	RESOURCE_PRECONDITION  = 1;
+	RESOURCE_POSTCONDITION = 2;
+	OUTPUT_PRECONDITION    = 3;
+}
+
+message ConditionResult {
+    // Address of the object containing the condition.
+    string addr = 1;
+
+    // Globally-unique address of the condition block.
+    string condition_addr = 2;
+
+    // Result indicates the value of the condition expression, which is
+    // true if the condition succeeds and false otherwise.
+    bool result = 3;
+
+    // Unknown, if true, indicates that the condition result can only be
+    // evaluated at apply time.
+    bool unknown = 4;
```

Comment:
Is the idea here that `result` is irrelevant if `unknown` is true? I guess that's fine but I wonder about putting the two inside of a [`oneof`](https://developers.google.com/protocol-buffers/docs/proto3#oneof) to represent that only one of them can be set at a time. (I guess that might make the generated Go code really awkward to work with though, so maybe not worth it...)

Reply:
That is the intent, although it was borne out of a desire to keep the mapping between plan data structured and file representation trivial. In the plan data structure I initially preferred to use a pair of bools to represent true/false/unknown, rather than a more permissive `cty.Value`.

I hadn't thought about using `oneof` here, and that makes sense to me, so I've updated both data types accordingly.